packages/scripts/src/migration/dvm2-designated-voting/common.ts

```diff
@@ -0,0 +1,141 @@
+import fetch from "node-fetch";
+
+import { VotingTokenEthers, DesignatedVotingFactoryEthers, DesignatedVotingEthers } from "@uma/contracts-node";
+import { getContractInstance } from "../../utils/contracts";
+import { BigNumber } from "ethers";
+
+interface EtherscanTransactionListResponse {
+  input: string;
+  from: string;
+  functionName: string;
+}
+
+export interface DesignatedVotingOwnerWithBalance {
+  designatedVoting: string;
+  owner: string;
+  voter: string;
+  balance: BigNumber;
+}
+
+export interface GnosisPayload {
+  version: string;
+  chainId: string;
+  createdAt: number;
+  meta: {
+    name: string;
+    description: string;
+    txBuilderVersion: string;
+    createdFromSafeAddress: string;
+    createdFromOwnerAddress: string;
+    checksum: string;
+  };
+  transactions: GnosisTransaction[];
+}
+
+export interface GnosisTransaction {
+  to: string;
+  value: string;
+  contractMethod: { inputs: { internalType: string; name: string; type: string }[]; name: string; payable: boolean };
+  contractInputsValues: { [key: string]: string };
+}
+
+export async function getDesignatedVotingContractsOwnedByOwner(
+  owner: string
+): Promise<DesignatedVotingOwnerWithBalance[]> {
+  const factoryV1 = await getContractInstance<DesignatedVotingFactoryEthers>("DesignatedVotingFactory");
+
+  const votingToken = await getContractInstance<VotingTokenEthers>("VotingToken");
+  // Step 1: find all DesignatedVoting contracts created by the DesignatedVotingFactory.
+  const designatedVotingContracts = await _fetchDesignatedVotingContractsCreatedByFactory(factoryV1);
+
+  // Step 3: return all DesignatedVoting contracts with the contract, owner, hot wallet and balance for each.
+  return (
+    await Promise.all(
+      designatedVotingContracts.map(async (DesignatedVotingAddress) => {
+        const contract = await getContractInstance<DesignatedVotingEthers>("DesignatedVoting", DesignatedVotingAddress);
+        const contractOwner = await contract.getMember(0);
+        const voter = await contract.getMember(1);
+        const balance = await votingToken.balanceOf(contract.address);
+        // Typescript knows these values are all defined.
+        return { designatedVoting: contract.address, owner: contractOwner, voter, balance };
+      })
+    )
+  ).filter((contract) => contract.owner === owner);
+}
+
+async function _fetchDesignatedVotingContractsCreatedByFactory(
```

Comment:
might have been easier to fetch internal transactions that deploy new contracts through factory: `action=txlistinternal` on Etherscan API and take `contractAddress` values for transactions with `create` type.
But this method should also work